ckanext/stcndm/logic/common.py

```diff
@@ -37,33 +30,47 @@ def get_next_product_id(context, data_dict):
     """
 
     product_id = _get_or_bust(data_dict, 'parentProductId')
+
+    # TODO: we need to standardize these. This is a result of swapping
+    # everything to java-style params
     data_dict['productId'] = product_id
-    # TODO: we need to standardize these. This is a result of swapping everything to java-style params
 
     if not len(str(product_id)) == 8:
         raise _ValidationError('invalid ParentProductId length')
 
     product_type = _get_or_bust(data_dict, 'productType')
 
-    _get_action('ndm_get_cube')(context, data_dict)  # testing for existance of cubeid
+    # testing for existance of cubeid
+    _get_action('ndm_get_cube')(context, data_dict)
```

Comment:
I strongly suggest switching to LocalCKAN for calling action methods. Using _get_action this way leaks context and data_dict modifications from one call to the next.